src/methods/properties.js

```diff
@@ -231,16 +231,23 @@ function ensureIsNotFinal(realm: Realm, O: ObjectValue, P: void | PropertyKeyVal
     return;
   }
 
-  // We can't continue because this object is already in its final state.
-  let error = new CompilerDiagnostic(
-    "Mutating a final object, or an object with unknown properties, after some of those " +
-      "properties have already been used, is not supported.",
-    realm.currentLocation,
-    "PP0026",
-    "FatalError"
-  );
-  realm.handleError(error);
-  throw new FatalError();
+  // We can't continue because this object is already in its final state
+  if (realm.instantRender.enabled) {
+    realm.instantRenderBailout(
+      "Object mutations that require materialization are currently not supported by InstantRender",
+      realm.currentLocation
+    );
+  } else {
+    let error = new CompilerDiagnostic(
+      "Mutating a final object, or an object with unknown properties, after some of those " +
```

Comment:
Not your doing, but I hate it when terms related to internal implementation details of Prepack leak out via error messages. Moreover, the text of this message is very obscure and may mislead a programmer who has to make sense of it. 

Even worse than that, the wiki page for this error gives an outdated example. Prepack already snapshots and does the right thing for the example. I'm none too sure if there are other cases. If there are none, the whole concept of final objects should be deleted. Otherwise, the wiki page should be updated with examples that cover ALL of the other cases.

Is there an issue for this?